### PR TITLE
Add documentation for version 1 of the Godot Android plugin.

### PR DESCRIPTION
**Pull request by m4gr3d**
Follow up documentation update for https://github.com/godotengine/godot/pull/33682

### CODE REVIEW

File: tutorials/plugins/android/android_plugin_v2.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+The gdnative resources can be bundled within the plugin ``aar`` file which simplifies the distribution and deployment process:
+
+    -   The shared libraries (``.so``) for the defined gdnative libraries will be automatically bundled by the ``aar`` build system. 
+
+    -   Godot ``*.gdnlib`` and ``*.gdns`` resource files must be manually defined in the plugin ``assets`` directory. 
+        The recommended path for these resources relative to the ``assets`` directory should be: ``godot/plugin/v2/[PluginName]/``.
+
+For gdnative libraries, the plugin singleton object must override the ``org.godotengine.godot.plugin.GodotPlugin::getPluginGDNativeLibrariesPaths()`` method, 
+and return the paths to the bundled gdnative libraries config files (``*.gdnlib``). The paths must be relative to the ``assets`` directory.
+At runtime, the plugin will provide these paths to Godot core which will use them to load and initialize the bundled gdnative libraries.
```
Comment (Calinou) [inline tutorials/plugins/android/android_plugin_v2.rst]:
```suggestion
At runtime, the plugin will provide these paths to Godot core which will use them to load and initialize the bundled GDNative libraries.
```

File: tutorials/plugins/android/android_plugin_v2.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+Troubleshooting
+---------------
+
+Godot crashes upon load
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Check ``adb logcat`` for possible problems, then:
+
+-  Check that the methods used in the Java singleton only use simple
+   Java datatypes, more complex ones are not supported.
```
Comment (Calinou) [inline tutorials/plugins/android/android_plugin_v2.rst]:
```suggestion
   Java datatypes. More complex datatypes are not supported.
```

File: tutorials/plugins/android/android_plugin.rst [hunk 3]
```diff
@@ -1,16 +1,16 @@
 .. _doc_android_plugin:
 
-Creating Android plugins
-========================
+Creating Android plugins (Godot 4.0+)
+=========================================
```
Comment (akien-mga) [inline tutorials/plugins/android/android_plugin.rst]: The underline should be the same length as the title (in mono font). Same for other titles below.

**Comment by Calinou** [comment]
@m4gr3d Is this PR now ready to merge?

**Comment by m4gr3d** [comment]
> @m4gr3d Is this PR now ready to merge, or are there upcoming changes to the main Godot repository that would make this documentation outdated?

@Calinou It's still awaiting completion of the review for PR #36336 by @akien-mga in case there is additional feedback to improve the documentation.
Once that's done, I'll squash the commit as well.

**Comment by m4gr3d** [comment]
@Calinou PR [#33682](https://github.com/godotengine/godot/pull/33682) was approved, so this PR should be ready for merging.

**Comment by Calinou** [comment]
Thanks!

**Pull request merged**